pilot/pkg/networking/core/route/route.go

```diff
@@ -1242,6 +1242,12 @@ func BuildDefaultHTTPInboundRoute(clusterName string, operation string) *route.R
 		// gRPC requests time out like any other requests using timeout or its default.
 		GrpcTimeoutHeaderMax: Notimeout,
 	}
+	out.GetRoute().RetryPolicy = &route.RetryPolicy{
+		RetryOn: "connect-failure,reset-before-request",
+		NumRetries: &wrapperspb.UInt32Value{
+			Value: 2,
```

Comment:
Do we really need 2? I thought the usecase was upstream closed the connection, which was long-lived, before we sent our request. A new request cannot possibly fail for the same reason?

Reply:
oh I guess it could try a different long-lived connection...

Reply:
That was my thought.